### ISSUE TITLE
FirebaseInAppMessaging code sign error on Xcode 14

### ISSUE THREAD

**Issue by minlite** [firebase/firebase-ios-sdk#9886]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 14.0 Beta (14A5228q)
  * Firebase SDK version: 9.1.0
  * Installation method: `Cocoapods`
  * Firebase Component: InAppMessaging 
  * Target platform(s): `iOS`

### [REQUIRED] Step 2: Describe the problem

When using Xcode 14 to build an iOS project containing the `InAppMessaging` component, the build fails with a code signing issue:

`Signing for "FirebaseInAppMessaging-InAppMessagingDisplayResources" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'FirebaseInAppMessaging-InAppMessagingDisplayResources' from project 'Pods'`

#### Steps to reproduce:

1. Add `FIrebaseInAppMessaging` pod to `Podfile`
2. Run `pod install`
3. Build iOS project in Xcode 14.
4. Error is shown:

`Signing for "FirebaseInAppMessaging-InAppMessagingDisplayResources" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'FirebaseInAppMessaging-InAppMessagingDisplayResources' from project 'Pods'`

### Quick fix

Selecting the team in the `Signing & Capabilities` tab of the `FirebaseInAppMessaging-InAppMessagingDisplayResources` project fixes the build.

**Comment by sarsonj** [comment]
This is not Firebase related bug - happens with other pods too. Workaround: https://github.com/CocoaPods/CocoaPods/issues/8891

**Comment by minlite** [comment]
@sarsonj I just checked again and it does seem this is a regression with Xcode 14 (even though the root cause might be the issue you mentioned).

## Xcode 13.4.1
1. Cloned project on our CI machine with Xcode 13.4.1 (13F100)
2. `pod install`
3. Build
4. Build is fine

## Xcode 14.0
1. Cloned project on a development machine with Xcode 14.0 Beta (14A5228q)
2. `pod install`
3. Build
4. Build fails with the error: `error build: Signing for "FirebaseInAppMessaging-InAppMessagingDisplayResources" requires a development team. Select a development team in the Signing & Capabilities editor.`

**Comment by SaloniSS** [comment]
@minlite Facing the same issue here

**Comment by ashutosh-akss** [comment]
facing same issue

**Comment by xiaogehenjimo** [comment]
facing same issue

**Comment by paulb777** [comment]
This is a CocoaPods/Xcode issue. See https://github.com/CocoaPods/CocoaPods/issues/11402.  Please add a reaction  instead of another comment, unless you have something to add about the workaround(s) or resolution.

**Comment by cdoky** [comment]
The Xcode14.0Beta2(14A522Pc) fixed the problem.

**Comment by cdoky** [comment]
The Xcode14.0Beta2 fixed the problem.

**Comment by yuanxun2007** [comment]
This post_install script in podfile fixed it. As it seems setting the own developer team is necessary. Replace Your Team ID with the TeamID of your project.

post_install do |installer|
  installer.generated_projects.each do |project|
    project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings["DEVELOPMENT_TEAM"] = "Your Team ID"
      end
    end
  end
end

**Comment by Tale-Dev** [comment]
> This post_install script in podfile fixed it. As it seems setting the own developer team is necessary. Replace Your Team ID with the TeamID of your project.
> 
> ```
> post_install do |installer|
>   installer.generated_projects.each do |project|
>     project.targets.each do |target|
>         target.build_configurations.each do |config|
>             config.build_settings["DEVELOPMENT_TEAM"] = "Your Team ID"
>          end
>     end
>   end
> end
> ```

Thanks @yuanxun2007, works for me.

**Comment by paulocoutinhox** [comment]
It works to me:
https://github.com/CocoaPods/CocoaPods/issues/8891

**Comment by wattanakorn495** [comment]
I have tried this script, it works fine on my local machine but it does not work on codemagic CI/CD
> This post_install script in podfile fixed it. As it seems setting the own developer team is necessary. Replace Your Team ID with the TeamID of your project.
> 
> ```
> post_install do |installer|
>   installer.generated_projects.each do |project|
>     project.targets.each do |target|
>         target.build_configurations.each do |config|
>             config.build_settings["DEVELOPMENT_TEAM"] = "Your Team ID"
>          end
>     end
>   end
> end
> ```

**Comment by yuanxun2007** [comment]
> I have tried this script, it works fine on my local machine but it does not work on codemagic CI/CD
> 
> > This post_install script in podfile fixed it. As it seems setting the own developer team is necessary. Replace Your Team ID with the TeamID of your project.
> > ```
> > post_install do |installer|
> >   installer.generated_projects.each do |project|
> >     project.targets.each do |target|
> >         target.build_configurations.each do |config|
> >             config.build_settings["DEVELOPMENT_TEAM"] = "Your Team ID"
> >          end
> >     end
> >   end
> > end
> > ```

please try bellow
target.build_configurations.each do |config|
          config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
 end

**Comment by mrazam110** [comment]
It Keeps failing on CI 
```
config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
or 
config.build_settings["DEVELOPMENT_TEAM"] = "Your Team ID"
```

Both not working

Any further help on this?

**Comment by paulb777** [comment]
Should be fixed with CocoaPods 1.12.0. See https://github.com/CocoaPods/CocoaPods/issues/11402
